### PR TITLE
feat: support spore nft

### PR DESCRIPTION
**Pull request by PainterPuppets**
reslove: https://github.com/Magickbase/ckb-explorer-public-issues/issues/395

Transaction page:

image

Collection list page

image

transfer:
image

inventory:
image

Collection info page

image

Collection item page

image

### CODE REVIEW

File: src/pages/NftInfo/index.tsx [hunk 1]
```diff
@@ -57,20 +67,42 @@ const NftInfo = () => {
   }
   const coverUrl = data?.data.icon_url || data?.data.collection.icon_url
 
-  return (
-    <div className={styles.container}>
-      <div className={styles.overview}>
-        {coverUrl ? (
+  const renderCover = () => {
+    if (data?.data.standard === 'spore_nft' && data?.data.cell && data?.data.cell.data) {
```
Comment (zhangyouxin) [inline src/pages/NftInfo/index.tsx:71]: many `data?.data.cell` here, better extract a variable

File: src/utils/spore.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { toBigEndian } from '@nervosnetwork/ckb-sdk-utils'
+import { hexToUtf8 } from './string'
+
+export function parseSporeClusterData(hexData: string) {
+  const data = hexData.slice(2, -1)
```
Comment (zhangyouxin) [inline src/utils/spore.ts:5]: could add a ref link here explaining why does it need to cut the tail

File: src/utils/spore.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  const nameOffset = Number(toBigEndian(`0x${data.slice(8, 8)}`)) * 2
```
Comment (zhangyouxin) [inline src/utils/spore.ts:7]: does `data.slice(8, 8)` not return empty string ?

File: src/utils/spore.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  const descriptionOffset = Number(toBigEndian(`0x${data.slice(16, 8)}`)) * 2
```
Comment (zhangyouxin) [inline src/utils/spore.ts:8]: `data.slice(16, 8)` is weird, `start` is bigger than `end`

File: src/utils/spore.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  const name = hexToUtf8(`0x${data.slice(nameOffset + 8, descriptionOffset)}`)
+  const description = hexToUtf8(`0x${data.slice(descriptionOffset + 8, -1)}`)
+
+  return { name, description }
+}
+
+export function parseSporeCellData(hexData: string) {
```
Comment (zhangyouxin) [inline src/utils/spore.ts:16]: maybe add some ref link here

File: src/__tests__/utils/spore.test.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { TextDecoder } from 'util'
+import { parseSporeCellData } from '../../utils/spore'
+
+const sporeCellData =
```
Comment (zhangyouxin) [inline src/__tests__/utils/spore.test.ts:4]: is there any explaination for this test case cell data

Comment (PainterPuppets) [reply src/__tests__/utils/spore.test.ts:4]: from this transfer spore cell transaction: [0x88be3c6ad1ca06dbecda108b539d288cfb134bdf4b65d7a14661a5f3bd045600](https://pudge.explorer.nervos.org/transaction/0x88be3c6ad1ca06dbecda108b539d288cfb134bdf4b65d7a14661a5f3bd045600)

File: src/components/NftCollectionTransfers/index.tsx [hunk 7]
```diff
@@ -101,21 +104,46 @@ const TransferTableRow: FC<{
...
+          src={`${patchMibaoImg(coverUrl)}?size=small&tid=${item.item.token_id}`}
+          alt="cover"
+          loading="lazy"
+          className={styles.icon}
+          onError={handleNftImgError}
+        />
+      )
+    }
+
+    return <img src="/images/nft_placeholder.png" alt="cover" loading="lazy" className={styles.icon} />
```
Comment (Keith-CY) [inline src/components/NftCollectionTransfers/index.tsx:139]: It can be simplified by updating `src` instead of returning different components

**Review by Keith-CY — APPROVED** [review]

**Review by Danie0918 — APPROVED** [review]
verified

**Pull request merged**